public/app/features/provisioning/utils/git.ts

```diff
@@ -0,0 +1,28 @@
+import { Repository } from '../api';
+
+/**
+ * Validates a Git branch name according to the following rules:
+ * 1. The branch name cannot start with `/`, end with `/`, `.`, or whitespace.
+ * 2. The branch name cannot contain consecutive slashes (`//`).
+ * 3. The branch name cannot contain consecutive dots (`..`).
+ * 4. The branch name cannot contain `@{`.
+ * 5. The branch name cannot include the following characters: `~`, `^`, `:`, `?`, `*`, `[`, `\`, or `]`.
+ * 6. The branch name must have at least one character and must not be empty.
+ */
+export function validateBranchName(branchName?: string) {
+  const branchNameRegex = /^(?!\/|.*\/\/|.*\.\.|.*@{)(?!.*[~^:?*[\]\\]).+(?<!\/|\.|\s)$/;
+
+  return branchName && branchNameRegex.test(branchName!);
+}
+
+export function getRemoteURL(repo: Repository) {
+  if (repo.spec?.type === 'github') {
+    const spec = repo.spec.github;
+    let url = spec?.url || '';
+    if (spec?.branch) {
+      url += `/tree/${spec.branch}`;
+    }
+    return url;
+  }
+  return undefined;
+}
```

Comment:
```suggestion
```
I don't think we need this anymore -- the resource endpoints should have the URL info (if appropriate).  We also can not let viewers see the full config